src/id128/id128.c

```diff
@@ -63,6 +66,112 @@ static int verb_invocation_id(int argc, char **argv, void *userdata) {
+        { "gpt-root-ia64-verity",      "GPT_ROOT_IA64_VERITY",      GPT_ROOT_IA64_VERITY   },
+
+        { "gpt-root-native",           "GPT_ROOT_NATIVE",           GPT_ROOT_NATIVE },
+#ifdef GPT_ROOT_SECONDARY
+        { "gpt-root-secondary",        "GPT_ROOT_SECONDARY",        GPT_ROOT_SECONDARY },
+#endif
+        { "gpt-root-native-verity",    "GPT_ROOT_NATIVE_VERITY",    GPT_ROOT_NATIVE_VERITY },
+#ifdef GPT_ROOT_SECONDARY_VERITY
+        { "gpt-root-secondary-verity", "GPT_ROOT_SECONDARY_VERITY", GPT_ROOT_SECONDARY_VERITY },
+#endif
+};
+
+static int gpt_table_lookup(const char *name, const gpt_table_entry **ret) {
+        for (size_t i = 0; i < ELEMENTSOF(gpt_table); i++)
+                if (streq(gpt_table[i].name, name)) {
+                        *ret = &gpt_table[i];
+                        return 0;
+                }
+
+        return -ENODATA;
+}
+
+static int show_one(Table **table, const gpt_table_entry *entry, bool first) {
+        int r;
+
+        if (arg_mode == ID128_PRINT_PRETTY) {
+                r = id128_pretty_print_sample(entry->cname, entry->uuid);
+                if (r < 0)
+                        return r;
+                if (!first)
+                        puts("");
+                return 0;
+
+        } else {
+                if (!*table) {
+                        *table = table_new("name", "uuid");
+                        if (!*table)
+                                return log_oom();
+                        table_set_width(*table, 0);
+                }
+
+                return table_add_many(*table,
+                                      TABLE_STRING, entry->name,
+                                      arg_mode == ID128_PRINT_ID128 ? TABLE_SD_128_T : TABLE_UUID,
+                                      entry->uuid);
+        }
+}
+
+static int verb_show(int argc, char **argv, void *userdata) {
+        _cleanup_(table_unrefp) Table *table = NULL;
+        char **p;
+        int r;
+
+        argv = strv_skip(argv, 1);
+        if (strv_isempty(argv))
+                for (size_t i = 0; i < ELEMENTSOF(gpt_table); i++) {
+                        r = show_one(&table, &gpt_table[i], i == 0);
+                        if (r < 0)
+                                return r;
+                }
+        else
+                STRV_FOREACH(p, argv) {
+                        const gpt_table_entry *e;
+
+                        r = gpt_table_lookup(*p, &e);
+                        if (r < 0)
+                                return log_error_errno(r, "Unknown id \"%s\".", *p);
+
+                        r = show_one(&table, e, p == argv);
+                        if (r < 0)
+                                return r;
+                }
+
+        if (table) {
+                r = table_print(table, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to print table: %m");
+        }
+
+        return 0;
+}
```

Comment:
hm, if this is just about the gpt partition tables, wouldn#t it be better to make this a verb of a tool associated with that, i.e. systemd-repart for example, see #14368. i.e. "systemd-repart --list-partition-types"?

Reply:
We could add it there too, but I think sd128 is a good home for it. Like mentioned above, I think we'll want to print other uuids in the future too.